Examples/AddNftAllowance/main.swift

```diff
@@ -0,0 +1,180 @@
+/*
+ * Hedera Swift SDK
+ *
+ * Copyright (C) 2022 - 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Hedera
+import SwiftDotenv
+
+@main
+internal enum Program {
+    internal static func main() async throws {
+        let env = try Dotenv.load()
+        let client = try Client.forName(env.networkName)
+
+        client.setOperator(env.operatorAccountId, env.operatorKey)
+
+        // Step 1: Create an NFT
+        let nftCreateReceipt = try await TokenCreateTransaction()
+            .name("HIP-336 NFT1")
+            .symbol("HIP336NFT1")
+            .tokenType(TokenType.nonFungibleUnique)
+            .decimals(0)
+            .initialSupply(0)
+            .maxSupply(10)
+            .tokenSupplyType(.finite)
+            .treasuryAccountId(env.operatorAccountId)
+            .adminKey(env.operatorKey.publicKey)
+            .supplyKey(env.operatorKey.publicKey)
+            .freezeWith(client)
+            .execute(client)
+            .getReceipt(client)
+
+        guard let nftTokenId = nftCreateReceipt.tokenId else {
+            fatalError("Failed to create NFT")
+        }
+
+        print("Created NFT with token ID: \(nftTokenId)")
+
+        // Step 2: Mint NFTs
+        let cids = [
+            "QmNPCiNA3Dsu3K5FxDPMG5Q3fZRwVTg14EXA92uqEeSRXn",
+            "QmZ4dgAgt8owvnULxnKxNe8YqpavtVCXmc1Lt2XajFpJs9",
+            "QmPzY5GxevjyfMUF5vEAjtyRoigzWp47MiKAtLBduLMC1T",
+        ]
+
+        let metadataArray = cids.map {
+            Data($0.utf8)
+        }
+
+        for cid in cids {
+            let mintReceipt = try await TokenMintTransaction()
+                .tokenId(nftTokenId)
+                .metadata(metadataArray)
+                .freezeWith(client)
+                .execute(client)
+                .getReceipt(client)
+
+            guard let serials = mintReceipt.serials, !serials.isEmpty else {
+                fatalError("Failed to mint NFTs")
+            }
+
+            for (index, serial) in serials.enumerated() {
+                print("Minted NFT (token ID: \(nftTokenId)) with serial: \(serials.first!)")
+            }
+
+        }
+
+        // Step 3: Create spender and receiver accounts
+        let spenderKey = PrivateKey.generateEd25519()
+        let receiverKey = PrivateKey.generateEd25519()
+
+        let spenderAccountId = try await AccountCreateTransaction()
+            .key(spenderKey.publicKey.single())
```

Comment:
Should be `Key.single(spenderKey.publicKey)` since the parameter requires type `Key`